### PR TITLE
Gesture state machine + test suite (stack 1/5)

### PR DESCRIPTION
**Pull request by joaodriessen**
First piece of the stack splitting up #23 (which is now a draft serving as the reference for the whole).

Adds the core of the touch-classification logic as a self-contained, I/O-free module (`src/runtime/gesture.rs`): it consumes raw evdev frames plus an explicit `Instant` and emits explicit outputs (relay to compositor / press / release / move) — no file descriptors, no wall clock. That's what makes every behavior testable without fingers on a pad.

The 30-test suite (`src/runtime/gesture/tests.rs`) encodes the failure modes live testing has hit: staggered touchdown/liftoff leaks (phantom taps and right-clicks), 4-finger transients through "exactly 3" in both directions, quick-tap verbatim replay (3-finger tap keeps working), drag-lock semantics, and SYN_DROPPED resync edge cases.

**Purely additive** — the module isn't wired into the runtime yet (that's stack 2) and no existing behavior changes. `cargo test` runs the suite.

Stack order: **1** → #25 → #26 → #27 → #28 (each builds on the previous; this one's diff is exactly its own slice).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/runtime/gesture.rs [hunk 1]
```diff
@@ -0,0 +1,745 @@
...
+    /// a slot release without tool release reads as a 2-finger tap:
+    /// phantom right-click).
+    clone_keys: Vec<(u16, i32)>,
+
+    /// Frames withheld from the compositor while a fresh touch is still
+    /// being classified.
+    pending: Vec<Ev>,
+    touch_start: Option<Instant>,
+    touch_max: usize,
+    settled: bool,
```
Comment (lmr97) [inline src/runtime/gesture.rs:178]: Can you elaborate on what information `settled` captures?

Comment (lmr97) [reply src/runtime/gesture.rs:178]: Never mind, looks like it is a way to keep track of when `probe_delay` has elapsed. Makes sense.

**Review by lmr97 — APPROVED** [review]
